core/build_playerglobal/src/lib.rs

```diff
@@ -62,3 +70,188 @@ pub fn build_playerglobal(
+            path += &flash_to_rust_path(resolve_multiname_ns(&abc, trait_name));
+            path += "::";
+        }
+
+        // Append the trait name - this corresponds to the actual method
+        // name (e.g. `getDefinitionByName`)
+        path += method_prefix;
+        path += resolve_multiname_name(&abc, trait_name);
+        println!("Rust path: {:?}", path);
+
+        // Now that we've built up the path, convert it into a `TokenStream`.
+        // This gives us something like
+        // `crate::avm2::globals::flash::system::Security::allowDomain`
+        //
+        // The resulting `TokenStream` is suitable for usage with `quote!` to
+        // generate a reference to the function pointer that should exist
+        // at that path in Rust code.
+        rust_paths.insert(method_id.0, TokenStream::from_str(&path).unwrap());
+    };
+
+    // We support three kinds of native methods:
+    // instance methods, class methods, and freestanding functions.
+    // We're going to insert them into an `FnvHashMap`, so the order
+    // that we visit them in doesn't matter.
+    for (i, instance) in abc.instances.iter().enumerate() {
+        // Look for native instance methods
+        for trait_ in &instance.traits {
+            check_trait(trait_, Some(instance.name));
+        }
+        // Look for native class methods (in the corresponding
+        // `Class` definition)
+        for trait_ in &abc.classes[i].traits {
+            check_trait(trait_, Some(instance.name));
+        }
+    }
+
+    // Look for freestanding methods
+    for script in &abc.scripts {
+        for trait_ in &script.traits {
+            check_trait(trait_, None);
+        }
+    }
+
+    let none = quote! { None };
+
+    // Finally, generate the actual code. This is a Rust array -
+    // the entry at index `i` is a Rust function pointer for the native
+    // method with id `i`. Not all methods in playerglobal will be native
+    // methods, so we store `None` in the entries corresponding to non-native
+    // functions. We expect the majority of the methods in playerglobal to be
+    // native, so this should only waste a small amount of memory.
+    //
+    // If a function pointer doesn't exist at the expected path,
+    // then Ruffle compilation will fail
+    // with an error message that mentions the non-existent path.
+    //
+    // When we initially load a method from an ABC file, we check if it's from our playerglobal,
+    // and if its ID exists in this table.
+    // If so, we replace it with a `NativeMethod` constructed
+    // from the function pointer we looked up in the table.
+
+    let table_methods = (0..abc.methods.len()).into_iter().map(|i| {
+        rust_paths
+            .get(&(i as u32))
+            .map_or(none.clone(), |path| quote! { Some(#path) })
+    });
+
+    let make_native_table = quote! {
+        const NATIVE_TABLE: &[Option<crate::avm2::method::NativeMethodImpl>] = &[
+            #(#table_methods),*
```

Comment:
nit: Change to `#(#table_methods,)*`, so that the last item will have a trailing comma.